### PR TITLE
[Benchmark] Fix ZeRO-3 step log

### PR DESCRIPTION
**Pull request by comaniac**
## Description ##
Fix the benchmark utility `train_with_torch` to consider micro batch when printing the log. Now it accepts optional `micro_batch_size` and only prints the loss per global batch.

## Checklist ##

- [x] PR's title starts with a category (e.g. [Bugfix], [Model], [Tutorial], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

cc @zarzen

### CODE REVIEW

File: examples/utils.py [hunk 1]
```diff
@@ -108,9 +114,10 @@ def train_with_torch(
             optimizer.step()
         loss = postproc(step, loss) if postproc is not None else loss
 
-        if step % 10 == 0:
-            logger.info(f"Step {step}, LOSS: {loss.item()}", ranks=0)
+        global_step = step // micro_batch_size
```
Comment (zarzen) [inline examples/utils.py:117]: The `step` is the index of the micro_batch, so `step // (global_batch_size // micro_batch_size)` is the right way to count the global step. Another way is leverage the runtime: 
```python
        if model.global_steps % 10 == 0 and model.is_gradient_accumulation_boundary():
            logger.info(f"Step {model.global_steps}, LOSS: {loss.item()}", ranks=0)

        loss = None
        if model.global_steps == steps:
            break
```

**Comment by comaniac** [comment]
Per offline discussion, we now have two functions to deal with different cases:
1. `train_with_deepspeed_engine`: Train with ZeRO but not pipeline. This function uses DeepSpeed model APIs `.global_steps` to print loss, so that we don't need to worry about batch size and DP size.
2. `train_with_torch`: Train with PyTorch native runtime. In this case we assume no DP, PP, and gradient accumulation, so each micro batch is just the global batch. This is currently only used by WideResNet w. TP.

In addition, this PR also reduces the number of steps in CI to reduce the CI time.

**Review by zarzen — APPROVED** [review]
